From 2b468b530cf369c068d13da01aba4bd35290d5d2 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 15:13:20 +0530
Subject: [PATCH 038/281] OpenWrt:
 465-m25p80-mx-disable-software-protection.patch

Disable software protection bits for Macronix flashes.

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: I127e2dc125acbdaa64f7e17e2c47dec65a94f888
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 drivers/mtd/spi-nor/macronix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mtd/spi-nor/macronix.c b/drivers/mtd/spi-nor/macronix.c
index d81a4cb2812b..94c6e793a7ac 100644
--- a/drivers/mtd/spi-nor/macronix.c
+++ b/drivers/mtd/spi-nor/macronix.c
@@ -106,6 +106,7 @@ static void macronix_nor_default_init(struct spi_nor *nor)
 {
 	nor->params->quad_enable = spi_nor_sr1_bit6_quad_enable;
 	nor->params->set_4byte_addr_mode = spi_nor_set_4byte_addr_mode;
+	nor->flags |= SNOR_F_HAS_LOCK;
 }
 
 static const struct spi_nor_fixups macronix_nor_fixups = {
-- 
2.17.1

